### ISSUE TITLE
Abandoned Guzzle Components

### ISSUE THREAD

**Issue by rowino** [KnpLabs/packagist-api#22]
The package uses abandoned guzzle components and needs upgrading:

```
Package guzzle/common is abandoned, you should avoid using it. Use guzzle/guzzle instead.
Package guzzle/stream is abandoned, you should avoid using it. Use guzzle/guzzle instead.
Package guzzle/parser is abandoned, you should avoid using it. Use guzzle/guzzle instead.
Package guzzle/http is abandoned, you should avoid using it. Use guzzle/guzzle instead.
```

**Comment by umpirsky** [comment]
@rowino Thanks for reporting this. I submitted a [fix](https://github.com/KnpLabs/packagist-api/pull/23) if you want to check before we merge it.

**Comment by pocallaghan** [comment]
Upon installing I get the message `guzzle/guzzle suggests installing guzzlehttp/guzzle (Guzzle 5 has moved to a new package name. The package you have installed, Guzzle 3, is deprecated.)`. I'm not sure if there are any BC issues to worry about if you switch to Guzzle 5, but sounds like Guzzle 3 is pretty old.

**Comment by umpirsky** [comment]
@pocallaghan Thanks for reoprting it.

I just installed and didn't see this message:

```
$ composer install
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
  - Installing doctrine/inflector (v1.0.1)
    Loading from cache

  - Installing symfony/event-dispatcher (v2.6.4)
    Loading from cache

  - Installing guzzle/guzzle (v3.9.2)
    Loading from cache

  - Installing symfony/yaml (v2.6.4)
    Loading from cache

	  - Installing symfony/process (v2.6.4)
    Loading from cache

  - Installing symfony/finder (v2.6.4)
    Loading from cache

  - Installing symfony/console (v2.6.4)
    Loading from cache

  - Installing sebastian/recursion-context (1.0.0)
    Loading from cache

  - Installing sebastian/exporter (1.2.0)
    Loading from cache

  - Installing phpdocumentor/reflection-docblock (2.0.4)
    Loading from cache

  - Installing doctrine/instantiator (1.0.4)
    Loading from cache

  - Installing phpspec/prophecy (v1.3.1)
    Loading from cache

  - Installing phpspec/php-diff (v1.0.2)
    Loading from cache

  - Installing phpspec/phpspec (2.1.1)
    Loading from cache

symfony/event-dispatcher suggests installing symfony/dependency-injection ()
symfony/event-dispatcher suggests installing symfony/http-kernel ()
symfony/console suggests installing psr/log (For using the console logger)
phpdocumentor/reflection-docblock suggests installing dflydev/markdown (~1.0)
phpdocumentor/reflection-docblock suggests installing erusev/parsedown (~1.0)
phpspec/phpspec suggests installing phpspec/nyan-formatters (~1.0 – Adds Nyan formatters)
Generating autoload files
```

Are you using latest master?

**Comment by pocallaghan** [comment]
@umpirsky Thanks for the quick response. I tried cloning the repo directly and as you say, the message doesn't appear. I created a test `composer.json` and added `packagist-api` as a dependency:

```javascript
{
    "name": "test/test",
    "type": "library",
    "minimum-stability": "dev",
    "require": {
      "knplabs/packagist-api": "master"
    }
}
```
Then running install for the project:
```
pocallaghan$ composer install
Loading composer repositories with package information
Installing dependencies (including require-dev)
  - Installing symfony/event-dispatcher (2.8.x-dev 15bbd5b)
    Cloning 15bbd5beed94cca89ffcce18fb76eeac38937240

  - Installing guzzle/guzzle (dev-master 0645b70)
    Cloning 0645b70d953bc1c067bbc8d5bc53194706b628d9

  - Installing doctrine/inflector (dev-master e5eaf8c)
    Cloning e5eaf8c7ded0877195b5d2848491e17b1c0a6c4d

  - Installing knplabs/packagist-api (dev-master 156bd51)
    Cloning 156bd5173838d61209a3011c64f9f048f3269f69

symfony/event-dispatcher suggests installing symfony/dependency-injection ()
symfony/event-dispatcher suggests installing symfony/http-kernel ()
guzzle/guzzle suggests installing guzzlehttp/guzzle (Guzzle 5 has moved to a new package name. The package you have installed, Guzzle 3, is deprecated.)
Writing lock file
Generating autoload files
```
I'm not massively *au fait* with composer, perhaps I'm doing something stupid in my test composer file. I tried playing around with the version I was requiring and the minimum-stability of the root project, but mainly received a lot of errors such as `The requested package knplabs/packagist-api could not be found in any version` when I did.

**Comment by pocallaghan** [comment]
@umpirsky I believe I've worked out the reason. Looks like your `composer.lock` file has guzzle as version `3.9.2`, but the latest release is actually `3.9.3` and this is the version the `suggest` data was added to `composer.json`. Since your `composer.json` file is `~3.0` anybody requiring `knplabs/packagist-api` will install `3.9.3` since this will ignore your `composer.lock` file.

Since version 3 is deprecated I guess the choices are:
 - Update the dependency to use Guzzle 5 (not sure how many BC breaks that would be that may require fixing).
 - Update the `composer.json` to point directly at `3.9.2`, knowing there will be no more updates to the deprecated code anyway.
 - Leave it as is and have anybody requiring the package see the suggestion.

Personally I think option 1 seems favourable, however it will obviously require the most work. If it's something your open to but don't have the time to commit, I could possible attempt it and look at making a pull request.

**Comment by umpirsky** [comment]
@pocallaghan Yes, I agree. Thanks for investigating this, really appreciate it. :+1: 

I would suggest going with 1. If the build pass, and examples work, we will go for it.

**Comment by GrahamCampbell** [comment]
> "knplabs/packagist-api": "master"

NEVER do that! Always use a version constraint.

**Comment by aimfeld** [comment]
@umpirsky If the fix works, please tag a new version, so we can install the new version.

**Comment by GrahamCampbell** [comment]
You don't need a new release. You can just use dev minimum stability.

**Comment by markushausammann** [comment]
You don't **need** a new release but it's best practice and good service to tag a fix, so people don't run the risk of installing whatever bugs you introduce in your master branch. It's generally recommended not to install dev-master in production but to always install tagged releases. The reasons are obvious. Of course in-house one may have master=stable workflow, but for libraries like this, the common best practice is semantic versioning. *Please tag a realease, it costs you nothing and helps everybody else.*

**Comment by markushausammann** [comment]
You know that you can tage releases directly from the github UI? You don't even need to break out the cli.

**Comment by GrahamCampbell** [comment]
Sure you can. You shouldn't pressure @umpirsky to tag though. There might be a good reason why he's not tagged yet.

**Comment by markushausammann** [comment]
After half a year? He just forgot to make a tag.

**Comment by umpirsky** [comment]
There is 1.3.0 now, I guess that helps?

**Comment by markushausammann** [comment]
Yes, thanks.

**Comment by markushausammann** [comment]
NO, actually it doesn't help because the 1.3.0 release still contains the old `guzzle/guzzle` namespace, please change to `guzzlehttp/guzzle` and publish a new release!

**Comment by GrahamCampbell** [comment]
> please change to guzzlehttp/guzzle and publish a new release!

No! That won't work. 3.x is deprecated either way, and changing to the other package is just plain wrong.

**Comment by GrahamCampbell** [comment]
Michael should really delete the 3.x tags from the new repo. The whole point of the new repo was to allow you to install 4+ alongside 3.

**Comment by markushausammann** [comment]
The point is that `guzzle/guzzle` is abandoned and will stop working at one point.

**Comment by Einenlum** [comment]
Cf. #25

**Comment by m1guelpf** [comment]
Fixed at #43

**Comment by MarioBlazek** [comment]
This one can be closed.
